### PR TITLE
Add CryptPad

### PR DESCRIPTION
**Pull request by sethforprivacy**

### CODE REVIEW

**Comment by sethforprivacy** [comment]
Two issues need to be resolved before this can be merged:

- httpUnsafeOrigin and httpSafeOrigin both need to be set, and currently cannot be without mounting a config file, which cannot be dynamic
- The data directory must be chown'd to 4001:4001 for Cryptpad to work, and we have no way to run simple scripts before starting containers ATM.

**Comment by meienberger** [comment]
Would this work if we create the initial `data` directory with 4001:4001 ?

**Comment by meienberger** [comment]
I mean if there is a data folder present besides the app config, it gets copied over upon installation. We should try to chown it to 4001:4001

**Comment by sethforprivacy** [comment]
Ohhh, I get it now.

Yes, I can give that a try and test it out.

**Comment by steveiliop56** [comment]
Hello @sethforprivacy, I will rework in the files completely...

File: apps/cryptpad/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - ${APP_DATA_DIR}/data/customize:/cryptpad/customize
+      - ${APP_DATA_DIR}/data/data:/cryptpad/data
+      - ${APP_DATA_DIR}/data/datastores:/cryptpad/datastore
+    labels:
+      traefik.enable: ${APP_EXPOSED}
+      traefik.http.routers.cryptpad.rule: Host(`${APP_DOMAIN}`)
+      traefik.http.routers.cryptpad.entrypoints: websecure
+      traefik.http.routers.cryptpad.service: cryptpad
+      traefik.http.routers.cryptpad.tls.certresolver: myresolver
+      traefik.http.services.cryptpad.loadbalancer.server.port: 80
```
Comment (steveiliop56) [inline apps/cryptpad/docker-compose.yml:33]: This needs to be replaced with the new traefik labels...

Comment (steveiliop56) [inline apps/cryptpad/docker-compose.yml:33]: This needs to be replaced with the new traefik labels...

File: apps/cryptpad/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    ulimits:
+      nofile:
+        soft: 1000000
+        hard: 1000000
+    volumes:
+      - ${APP_DATA_DIR}/data/blob:/cryptpad/blob
+      - ${APP_DATA_DIR}/data/block:/cryptpad/block
+      - ${APP_DATA_DIR}/data/customize:/cryptpad/customize
+      - ${APP_DATA_DIR}/data/data:/cryptpad/data
+      - ${APP_DATA_DIR}/data/datastores:/cryptpad/datastore
```
Comment (steveiliop56) [inline apps/cryptpad/docker-compose.yml:26]: You could just mount ``/cryptpad`` to ``- ${APP_DATA_DIR}/data`` directly...

File: apps/cryptpad/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+version: "3.7"
+services:
+  cryptpad:
+    image: promasu/cryptpad:nginx
```
Comment (steveiliop56) [inline apps/cryptpad/docker-compose.yml:4]: it would be better to use a version.

File: apps/cryptpad/config.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  "$schema": "../schema.json",
+  "name": "CryptPad",
+  "available": true,
+  "exposable": true,
+  "port": 8108,
+  "id": "cryptpad",
+  "tipi_version": 1,
+  "version": "latest",
+  "categories": ["utilities"],
+  "description": "CryptPad is a collaboration suite that is end-to-end-encrypted and open-source. It is built to enable collaboration, synchronizing changes to documents in real time. Because all data is encrypted, the service and its administrators have no way of seeing the content being edited and stored.",
```
Comment (steveiliop56) [inline apps/cryptpad/config.json:11]: That is too big and it will not look welll...

File: apps/cryptpad/config.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+  "$schema": "../schema.json",
+  "name": "CryptPad",
+  "available": true,
+  "exposable": true,
+  "port": 8108,
+  "id": "cryptpad",
+  "tipi_version": 1,
+  "version": "latest",
```
Comment (steveiliop56) [inline apps/cryptpad/config.json:9]: Here it should have the same version as the image in the docker compose file.

File: apps/cryptpad/config.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
+  "$schema": "../schema.json",
+  "name": "CryptPad",
+  "available": true,
+  "exposable": true,
+  "port": 8108,
```
Comment (steveiliop56) [inline apps/cryptpad/config.json:6]: This port is in use right now...

File: apps/cryptpad/metadata/description.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+![CryptPad screenshot](https://github.com/xwiki-labs/cryptpad/blob/main/screenshot.png?raw=true "Private real-time collaboration on a Rich Text document.")
+
+## Installation
+
+***NOTE: In order to properly function, you must correct permissions on the app-data directory for Cryptpad by running the following command from the runtipi folder:***
+
+`sudo chown -R 4001:4001 app-data/cryptpad`
+
+After running the above command, restart the app.
```
Comment (steveiliop56) [inline apps/cryptpad/metadata/description.md:13]: We will recheck this but right now it should be in the top of the readme file.

File: apps/cryptpad/metadata/logo.jpg
Comment (steveiliop56) [inline apps/cryptpad/metadata/logo.jpg:1]: This definitely needs to  be replaced....

**Review by steveiliop56 — CHANGES_REQUESTED** [review]

**Comment by steveiliop56** [comment]
Closing for inactivity...

**Pull request closed without merging**